js/src/client.ts

```diff
@@ -874,19 +900,73 @@ export class Client implements LangSmithTracingClientInterface {
       ...run,
       start_time: run.start_time ?? Date.now(),
     });
+
     if (
-      this.autoBatchTracing &&
       runCreate.trace_id !== undefined &&
       runCreate.dotted_order !== undefined
     ) {
-      void this.processRunOperation({
-        action: "create",
-        item: runCreate,
-      }).catch(console.error);
-      return;
+      if (this._rustClient) {
+        const mergedRunCreateParam = mergeRuntimeEnvIntoRunCreate(runCreate);
+
+        // We need to massage the data shape into what Rust expects and will accept,
```

Comment:
Is there any way we can remove this shim in the JS client?

Feels like it belongs in the Rust binding?